gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java

```diff
@@ -63,6 +63,7 @@ private ManagedChannel buildManagedChannel() {
         .createChannelBuilder(target)
         .enableRetry()
         .userAgent(options.getAppName())
+        .intercept(new InvocationIdGrpcChannelInterceptor())
```

Comment:
can you add a comment explaining why we are adding a Channel interceptor and not a stub interceptor?

Reply:
Adding InvocationId to header is generic thing which will apply to any RPC call made from gRPC client. That's the reason it's not added to a specific stub which could differ for different RPCs.

Reply:
Can we add this as a comment in the code since this is not trivial and would help future readers of the code.